### ISSUE TITLE
Package versions have conflicting dependencies.

### ISSUE THREAD

**Issue by Strawberry9583** [automl/DACBench#136]
Hi,
When using `pip install dacbench[sgd]` to install the additional dependency, I had conflicting dependencies and failed to use your benchmark.

My environment:
`Python 3.10.12`

My reported issue:
```ERROR: Cannot install dacbench, dacbench[all,dev,docs,example]==0.2.0 and ray[rllib]==2.3.1 because these package versions have conflicting dependencies.

The conflict is caused by:
    dacbench[all,dev,docs,example] 0.2.0 depends on numpy==1.24.2
    dacbench[all,dev,docs,example] 0.2.0 depends on numpy==1.24.2; extra == "all"
    cma 3.3.0 depends on numpy
    coax 0.1.13 depends on numpy>=1.21.6
    configspace 0.6.1 depends on numpy
    gymnasium 0.26.3 depends on numpy>=1.18.0
    matplotlib 3.7.1 depends on numpy>=1.20
    modcma 0.0.2.8.4 depends on numpy
    pandas 1.5.3 depends on numpy>=1.21.0; python_version >= "3.10"
    ray[rllib] 2.3.1 depends on numpy>=1.19.3; python_version >= "3.9"
    scikit-learn 1.2.2 depends on numpy>=1.17.3
    scipy 1.10.1 depends on numpy<1.27.0 and >=1.19.5
    seaborn 0.12.2 depends on numpy!=1.24.0 and >=1.17
    tensorflow 2.12.0 depends on numpy<1.24 and >=1.22

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```
To reproduce the issue, you can just follow the pip procedure in the GitHub repo directly, following your recommended [command lines](https://github.com/automl/DACBench#:~:text=Instead%20of%20using,recursive%0Apip%20install%20.).

Thank you very much for your consideration. I am looking forward to hearing back from you.

Best regards,
Hai

**Comment by TheEimer** [comment]
Thanks for pointing this out, unfortunately something about the dependencies don't seem to resolve correctly on PyPI, as you can see the command tries to install all dependencies at once which leads to issues due to different versioning of the target domains. I don't exactly know why this is happening, but will look into it. In the meantime, I recommend installing from GitHub. Sorry, I hope we can resolve this soon!

**Comment by TheEimer** [comment]
Just released 0.3.0 which for me installs from PyPI again. We also recommend uv now which also makes this easier. I'll close for now, feel free to re-open it the issue persists on you side.
